### PR TITLE
SRE-2519: New LICENSE file as expected by OpenSSF Scorecard

### PR DESCRIPTION
**Pull request by osalyk**
Doc-only: true

### CODE REVIEW

**Review by grom72 — DISMISSED** [review]

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (C) 2019-2024 Intel Corporation
```
Comment (janekmi) [inline Makefile]:
```suggestion
# Copyright (c) 2022-2024 Intel Corporation
```

File: Jenkinsfile [hunk 2]
```diff
@@ -1,41 +1,7 @@
...
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
- * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
- * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
- * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
- * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
- * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+// Copyright (c) 2019-2024 Intel Corporation
```
Comment (janekmi) [inline Jenkinsfile:3]: It looks like a copy-paste error. 🤔 

```suggestion
// Copyright (c) 2022-2024 Intel Corporation
```

**Review by janekmi — CHANGES_REQUESTED** [review]

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,54 @@
+BSD-2-Clause Plus Patent License
+
+Copyright 2019-2024 Intel Corporation.
```
Comment (janekmi) [inline LICENSE]:
```suggestion
Copyright 2022-2024 Intel Corporation.
```

**Review by janekmi — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (osalyk) [reply Makefile]: Done

**Review by grom72 — DISMISSED** [review]

**Review by janekmi — DISMISSED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Please rebase this PR on d14a57b2d7df31cd0261fbaa701630cf3f230754 so that we remove the archiving notice commit.

**Comment by osalyk** [comment]
> Please rebase this PR on [d14a57b](https://github.com/daos-stack/fuse/commit/d14a57b2d7df31cd0261fbaa701630cf3f230754) so that we remove the archiving notice commit.

Done.

**Review by janekmi — APPROVED** [review]

**Review by grom72 — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> Done.

Doesn't seem to be quite right.  You have taken the path to add a commit to your PR to revert 770ea4599b7a0eda28c6abf12a8129a1bf2b1f66.  We only want to do that when the _Archiving Notice_ commit is not the most recent commit on the branch (as happens inadvertently, unfortunately).  What I would like to see instead is the result of:

```
$ git rebase -i c86b3516e3ca2d1b1a4c8d66f3299389c8bff833
```

and then in the editor delete the _Revert "Archiving Notice"_ commit and then save.  What you should get as a result is the following commit history:

```
improve Makefile
improve Jenkinsfile
SRE-2519: New LICENSE file as expected by OpenSSF Scorecard
Merge upstream to get features required (#2)
Initial commit (#1)
Initial commit
```

So as you can see, you are completely removing the [_Archiving Notice_](c86b3516e3ca2d1b1a4c8d66f3299389c8bff833) commit from the history.

Once you have that, force push it to this PR.

**Comment by janekmi** [comment]
> @brianjmurrell wrote:
>
> So as you can see, you are completely removing the [_Archiving Notice_](c86b3516e3ca2d1b1a4c8d66f3299389c8bff833) commit from the history.
> 
> Once you have that, force push it to this PR.

1. Do you plan to force-push it to the master branch as well? As far as I know, to remove a commit from the history you have to overwrite the history. A regular PR landing won't do the trick, will it?
2. Why do you argue that this commit should be removed from history?
3. If you think removing this commit from the history is worth doing and if it requires overwriting the history anyways why won't you just do it in parallel to this pull request?

**Comment by grom72** [comment]
> > Done.
> 
> Doesn't seem to be quite right. You have taken the path to add a commit to your PR to revert [770ea45](https://github.com/daos-stack/fuse/commit/770ea4599b7a0eda28c6abf12a8129a1bf2b1f66). We only want to do that when the _Archiving Notice_ commit is not the most recent commit on the branch (as happens inadvertently, unfortunately). What I would like to see instead is the result of:
> 
> ```
> $ git rebase -i c86b3516e3ca2d1b1a4c8d66f3299389c8bff833
> ```
> 
> and then in the editor delete the _Revert "Archiving Notice"_ commit and then save. What you should get as a result is the following commit history:
> 
> ```
> improve Makefile
> improve Jenkinsfile
> SRE-2519: New LICENSE file as expected by OpenSSF Scorecard
> Merge upstream to get features required (#2)
> Initial commit (#1)
> Initial commit
> ```
> 
> So as you can see, you are completely removing the [_Archiving Notice_](c86b3516e3ca2d1b1a4c8d66f3299389c8bff833) commit from the history.
> 
> Once you have that, force push it to this PR.

@brianjmurrell 
But how it's supposed to work?
I have such a branch https://github.com/daos-stack/fuse/commits/grom72/LICENSE/
but it does not remove [_Archiving Notice_](c86b3516e3ca2d1b1a4c8d66f3299389c8bff833)  when creating a PR->master based on that.
The only way is to make a forced push to master (as far as I know).

**Comment by brianjmurrell** [comment]
> 1. Do you plan to force-push it to the master branch as well?

Yes.

> As far as I know, to remove a commit from the history you have to overwrite the history. A regular PR landing won't do the trick, will it?

It will if it is (what we call) _force-landed_.

> 2. Why do you argue that this commit should be removed from history?

Because we don't really even want it there in the first place but we have no choice.  This is being forced on us and is a bad process IMO, to just assume that software that is not updated MUST be insecure, vulnerable or abandoned.  There is such a thing as stable software and this process is not allowing for that.

> 3. If you think removing this commit from the history is worth doing and if it requires overwriting the history anyways why won't you just do it in parallel to this pull request?

Because it can all be done all in one single operation with the PR.  If we did do them separately as you suggest. the PR will need to be rebased once the archiving commit has been force-pushed off of history anyway.  All of that makes for more work for everyone.

**Comment by brianjmurrell** [comment]
> @brianjmurrell But how it's supposed to work? I have such a branch [`grom72`/LICENSE/ (commits)](https://github.com/daos-stack/fuse/commits/grom72/LICENSE/)

Yes.

> but it does not remove [_Archiving Notice_](c86b3516e3ca2d1b1a4c8d66f3299389c8bff833) when creating a PR->master based on that.

Ideally when you initially created your branch, you would have branched from d14a57b2d7df31cd0261fbaa701630cf3f230754 (master~1) rather than the tip of master.  But since it was done from master, it just needs a rebase now.

> The only way is to make a forced push to master (as far as I know).

Which is what will happen when it's time to land this PR.

**Comment by brianjmurrell** [comment]
So to be perfectly clear, yes, we are rewriting history.  But that's because we are removing history that we don't believe should really be there.  Somebody else created it for us against our judgement.  So we remove it.

**Comment by grom72** [comment]
> So to be perfectly clear, yes, we are rewriting history. But that's because we are removing history that we don't believe should really be there. Somebody else created it for us against our judgement. So we remove it.

Done

**Comment by janekmi** [comment]
> So to be perfectly clear, yes, we are rewriting history. But that's because we are removing history that we don't believe should really be there. Somebody else created it for us against our judgement. So we remove it.

Thanks for the clarification. Would it make any difference if we just commit a revert of the unwanted commit and continue business without re-writing the history?

**Comment by brianjmurrell** [comment]
> Thanks for the clarification. Would it make any difference if we just commit a revert of the unwanted commit and continue business without re-writing the history?

Functionally, no.  But it does muddy up the history with "project abandonment" commits that are not actually true.  So to remove any confusion they cause, we remove the commits.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**